dkc/settings.py

```diff
@@ -10,6 +11,22 @@
     ProductionBaseConfiguration,
     TestingBaseConfiguration,
 )
+from django.core.validators import MaxLengthValidator, RegexValidator
+
+
+class UsernameRegexValidator(RegexValidator):
+    regex = r'^[a-z\d](?:[a-z\d]|-(?=[a-z\d]))*$'
```

Comment:
Why subclass instead of passing these as arguments to the constructor?

Reply:
I just copied how [these](https://github.com/django/django/blob/master/django/contrib/auth/validators.py) were done. I have no preference, I'm happy to change it if you want.